### PR TITLE
Domains: Simplify single site domains page.

### LINKED ISSUE #84992: Domains: Simplify screen when there's only 1 domain
### Details

On the single site domains search page, we should simplify by removing search, remove illustration, remove manage all domains if only 1 domain.

After:

image

### Checklist

- [x] Search field doesn't show up if there's only one domain.
- [x] Manage All Domains doesn't show up if the user only has 1 site.
- [x] Manage All Domains doesn't show up if the user has more than 1 site but only 1 domain.
- [x] Update the "Get your domain" title to include "Free".

### Related

p58i-fXA-p2

### PR DESCRIPTION
**Pull request by yansern**
Fixes https://github.com/Automattic/wp-calypso/issues/84992

## Proposed Changes

On single site domains page:
 - Don't show search field if there's only one domain _(subdomains not part of it)_.
 - Don't show checkbox column if site has only one domain _(subdomains not part of it)_.
 - Don't show Manage All Domains CTA if the user has only one domain _(subdomains not part of it)_.
 - Update the "Get your domain" in Get Your Free Domain CTA to "Get your free domain".
 - Remove illustration on Get Your Free Domain CTA.

## Testing Instructions

**Using the link `/domains/manage/your-test-site.wordpress.com`:**
* On a simple site that has a subdomain, it should match all of the above proposed changes.
* On a simple site that has a subdomain and a domain, it should match all of the above proposed changes.
* On a simple site that has a subdomain and 2 domains and above, it should only match the last 2 proposed changes above.

**Test regression at `/domains/manage`:**
* There should be no regressions on the All Domains page.

## Screenshot

| Before | After |
| --- | --- |
| image | image |
| image | image |

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/domains/domain-management/list/empty-domains-list-card.jsx [hunk 1]
```diff
@@ -20,7 +26,7 @@ function EmptyDomainsListCard( { selectedSite, hasDomainCredit, isCompact, hasNo
 
 	const siteHasHundredYearPlan = selectedSite?.plan?.product_slug === PLAN_100_YEARS;
 
-	let title = translate( 'Get your domain' );
+	let title = translate( 'Get your free domain' );
```
Comment (phcp) [inline client/my-sites/domains/domain-management/list/empty-domains-list-card.jsx]: It would be nice to add a translation check here.
```
	let title = i18n.hasTranslation( 'Get your free domain' )
		? translate( 'Get your free domain' )
		: translate( 'Get your domain' );
```

**Review by phcp — APPROVED** [review]
Nice work! Changes are working as expected, and I didn't notice any regression. Just left one suggestion, but feel free to move forward.

File: client/my-sites/domains/domain-management/list/empty-domains-list-card.jsx (hunk 1 above)
Comment (obenland) [reply client/my-sites/domains/domain-management/list/empty-domains-list-card.jsx]: Agreed, that's a quick quality win.

File: client/my-sites/domains/domain-management/list/empty-domains-list-card-skeleton.tsx [hunk 2]
```diff
@@ -55,7 +57,9 @@ export const EmptyDomainsListCardSkeleton = ( {
 					'has-title-only': title && ! line,
 				} ) }
 			>
-				<div className="empty-domains-list-card__illustration">{ illustration }</div>
+				{ showIllustration && (
+					<div className="empty-domains-list-card__illustration">{ illustration }</div>
+				) }
```
Comment (obenland) [inline client/my-sites/domains/domain-management/list/empty-domains-list-card-skeleton.tsx]: Let's remove the illustration entirely.

**Review by obenland — DISMISSED** [review]

File: client/my-sites/domains/domain-management/list/empty-domains-list-card-skeleton.tsx (hunk 2 above)
Comment (yansern) [reply client/my-sites/domains/domain-management/list/empty-domains-list-card-skeleton.tsx]: Done.

File: client/my-sites/domains/domain-management/list/bulk-site-domains.tsx [hunk 3]
```diff
@@ -82,6 +82,10 @@ export default function BulkSiteDomains( props: BulkSiteDomainsProps ) {
 	const [ changeSiteAddressSourceDomain, setChangeSiteAddressSourceDomain ] =
 		useState< ResponseDomain | null >( null );
 
+	// If a site contains more than one domain that is not a subdomain.
+	const showManageAllDomainsCTA =
+		( data?.domains ?? [] ).filter( ( domain ) => ! domain.is_subdomain ).length > 1;
```
Comment (obenland) [inline client/my-sites/domains/domain-management/list/bulk-site-domains.tsx:87]: @yansern Could you take another pass at this? The Manage All Domains button should show if there are domains on any other site.

Comment (yansern) [reply client/my-sites/domains/domain-management/list/bulk-site-domains.tsx:87]: Just to verify, do you mean this?
 - If a site has 1 domain (excluding subdomains), it shouldn't show.
 - If a site has 2 or more domains (excluding subdomains), it should show.

Comment (obenland) [reply client/my-sites/domains/domain-management/list/bulk-site-domains.tsx:87]: I meant this
* Manage All Domains doesn't show up if the user only has 1 site.
* Manage All Domains doesn't show up if the user has more than 1 site but only 1 domain.

Maybe the second one should be
* Manage All Domains doesn't show up if the user has more than 1 site but only 1 domain across those sites.

The Manage All Domains screen only makes sense for cases where users have more than one domains on more than one site.

Comment (yansern) [reply client/my-sites/domains/domain-management/list/bulk-site-domains.tsx:87]: Okay gotcha, let me try again.

Comment (yansern) [reply client/my-sites/domains/domain-management/list/bulk-site-domains.tsx:87]: Followed up here. https://github.com/Automattic/wp-calypso/pull/85414
